Source/SwiftLintFramework/Rules/Lint/NSLocalizedStringKeyRule.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import SourceKittenFramework
+
+public struct NSLocalizedStringKeyRule: ASTRule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "nslocalizedstring_key",
+        name: "NSLocalizedString Key",
+        description: "Static strings should be used as key in NSLocalizedString in order to genstrings work.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            "NSLocalizedString(\"key\", comment: nil)",
+            "NSLocalizedString(\"key\" + \"2\", comment: nil)"
+        ],
+        triggeringExamples: [
+            "NSLocalizedString(↓method(), comment: nil)",
+            "NSLocalizedString(↓\"key_\\(param)\", comment: nil)"
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .call,
+            dictionary.name == "NSLocalizedString",
+            let firstArgument = dictionary.enclosedArguments.first,
+            firstArgument.name == nil,
+            let offset = firstArgument.offset,
+            let length = firstArgument.length,
+            case let kinds = file.syntaxMap.kinds(inByteRange: NSRange(location: offset, length: length)),
+            Set(kinds) != [.string] else {
```

Comment:
Just curious what you think of `!kinds.allSatisfy({ $0 == .string })` instead? More efficient since it can return early instead of iterating through the whole sequence and doesn't have to compute the hash for each member of `kinds`.

Reply:
Agreed it shouldn't make much of a difference in this case.